app/lib/actions/katello/host/unregister.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Katello
+    module Host
+      class Unregister < Actions::EntryAction
+        middleware.use ::Actions::Middleware::RemoteAction
+
+        def plan(host, options = {})
+          plan_action(Katello::Host::Destroy, host, options.merge(:destroy_object => false))
```

Comment:
The Host::Destroy takes care of that.  The purpose of this action is the fact that for an 'unregister' api call, we don't actually want to destroy the host, just 'unregister' it from candlepin and pulp`